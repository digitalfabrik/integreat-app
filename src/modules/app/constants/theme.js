import colors from './colors'
import dimensions from './dimensions'
import fonts from './fonts'
import helpers from './helpers'

export default {
  colors,
  dimensions,
  fonts,
  helpers
}
