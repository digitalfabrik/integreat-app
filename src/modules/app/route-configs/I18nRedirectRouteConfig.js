// @flow

import { RouteConfig } from './RouteConfig'
import { citiesEndpoint, Payload, CityModel } from '@integreat-app/integreat-api-client'
import type { Dispatch, GetState, Route } from 'redux-first-router'
import fetchData from '../fetchData'
import type { AllPayloadsType } from './RouteConfig'

type I18nRedirectRouteParamsType = {|param?: string|}
type RequiredPayloadsType = {|cities: Payload<Array<CityModel>>|}

export const I18N_REDIRECT_ROUTE = 'I18N_REDIRECT'

/**
 * I18nRoute to redirect if no language is specified or to the not found route if the param is invalid.
 * Matches / and /param
 * @type {{path: string, thunk: function(Dispatch, GetState)}}
 */
const i18nRedirectRoute: Route = {
  path: '/:param?',
  thunk: async (dispatch: Dispatch, getState: GetState) => {
    const state = getState()

    await fetchData(citiesEndpoint, dispatch, state.cities)
  }
}

class I18nRedirectRouteConfig implements RouteConfig<I18nRedirectRouteParamsType, RequiredPayloadsType> {
  name = I18N_REDIRECT_ROUTE
  route = i18nRedirectRoute
  isLocationLayoutRoute = false
  requiresHeader = false
  requiresFooter = false

 getRoutePath = ({param}: I18nRedirectRouteParamsType): string => `/${param || ''}`

  getRequiredPayloads = (payloads: AllPayloadsType): RequiredPayloadsType => ({cities: payloads.citiesPayload})

  getPageTitle = () => ''

  getLanguageChangePath = () => null

  getMetaDescription = () => null

  getFeedbackReference = () => null
}

export default I18nRedirectRouteConfig
