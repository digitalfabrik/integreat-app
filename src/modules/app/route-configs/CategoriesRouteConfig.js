// @flow

import type { AllPayloadsType } from './RouteConfig'
import { RouteConfig } from './RouteConfig'
import type { Route } from 'redux-first-router'
import fetchData from '../fetchData'
import {
  CategoriesMapModel,
  CityModel,
  createCategoriesEndpoint,
  createCitiesEndpoint,
  createEventsEndpoint,
  createLocalNewsEndpoint,
  createLanguagesEndpoint,
  Payload
} from '@integreat-app/integreat-api-client'
import { cmsApiBaseUrl } from '../constants/urls'

export type CategoriesRouteParamsType = {|city: string, language: string|}
type RequiredPayloadsType = {|categories: Payload<CategoriesMapModel>, cities: Payload<Array<CityModel>>|}

export const CATEGORIES_ROUTE = 'CATEGORIES'

/**
 * CategoriesRoute, matches /augsburg/de*
 * @type {{path: string, thunk: function(Dispatch, GetState)}}
 */
const categoriesRoute: Route = {
  path: '/:city/:language/:categoryPath*',
  thunk: async (dispatch, getState) => {
    const state = getState()
    const { city, language } = state.location.payload

    await Promise.all([
      fetchData(createCitiesEndpoint(cmsApiBaseUrl), dispatch, state.cities),
      fetchData(createEventsEndpoint(cmsApiBaseUrl), dispatch, state.events, { city, language }),
      fetchData(createLocalNewsEndpoint(cmsApiBaseUrl), dispatch, state.news, { city, language }),
      fetchData(createLanguagesEndpoint(cmsApiBaseUrl), dispatch, state.languages, { city, language }),
      fetchData(createCategoriesEndpoint(cmsApiBaseUrl), dispatch, state.categories, { city, language })
    ])
  }
}

class CategoriesRouteConfig implements RouteConfig<CategoriesRouteParamsType, RequiredPayloadsType> {
  name = CATEGORIES_ROUTE
  route = categoriesRoute
  isLocationLayoutRoute = true
  requiresHeader = true
  requiresFooter = true

  getRoutePath = ({ city, language }: CategoriesRouteParamsType): string => `/${city}/${language}`

  getRequiredPayloads = (payloads: AllPayloadsType): RequiredPayloadsType =>
    ({ categories: payloads.categoriesPayload, cities: payloads.citiesPayload })

  getLanguageChangePath = ({ location, payloads, language }) => {
    const { city } = location.payload
    const categories = payloads.categories.data
    if (categories) {
      const category = categories.findCategoryByPath(location.pathname)
      if (category && !category.isRoot()) {
        return category.availableLanguages.get(language) || null
      }
    }
    return this.getRoutePath({ city, language })
  }

  getPageTitle = ({ t, payloads, cityName, location }) => {
    if (!cityName) {
      return null
    }
    const pathname = location.pathname
    const categories = payloads.categories.data
    const category = categories && categories.findCategoryByPath(pathname)
    return `${category && !category.isRoot() ? `${category.title} - ` : ''}${cityName}`
  }

  getMetaDescription = t => t('metaDescription')

  getFeedbackTargetInformation = ({ location, payloads }) => {
    const categories = payloads.categories.data
    const category = categories && categories.findCategoryByPath(location.pathname)
    return category ? { path: category.path, title: category.title } : null
  }
}

export default CategoriesRouteConfig
