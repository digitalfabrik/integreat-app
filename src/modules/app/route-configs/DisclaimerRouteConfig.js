// @flow

import type { AllPayloadsType } from './RouteConfig'
import { RouteConfig } from './RouteConfig'
import type { Route } from 'redux-first-router'
import fetchData from '../fetchData'
import {
  createCitiesEndpoint,
  createDisclaimerEndpoint,
  createEventsEndpoint,
  createLanguagesEndpoint,
  PageModel,
  Payload
} from '@integreat-app/integreat-api-client'
import { cmsApiBaseUrl } from '../constants/urls'

type DisclaimerRouteParamsType = {|city: string, language: string|}
type RequiredPayloadsType = {|disclaimer: Payload<PageModel>|}

export const DISCLAIMER_ROUTE = 'DISCLAIMER'

/**
 * DisclaimerRoute (for city specific disclaimers), matches /augsburg/de/disclaimer
 * @type {{path: string, thunk: function(Dispatch, GetState)}}
 */
const disclaimerRoute: Route = {
  path: '/:city/:language/disclaimer',
  thunk: async (dispatch, getState) => {
    const state = getState()
    const {city, language} = state.location.payload

    await Promise.all([
      fetchData(createCitiesEndpoint(cmsApiBaseUrl), dispatch, state.cities),
      fetchData(createEventsEndpoint(cmsApiBaseUrl), dispatch, state.events, {city, language}),
      fetchData(createLanguagesEndpoint(cmsApiBaseUrl), dispatch, state.languages, {city, language}),
      fetchData(createDisclaimerEndpoint(cmsApiBaseUrl), dispatch, state.disclaimer, {city, language})
    ])
  }
}

class DisclaimerRouteConfig implements RouteConfig<DisclaimerRouteParamsType, RequiredPayloadsType> {
  name = DISCLAIMER_ROUTE
  route = disclaimerRoute
  isLocationLayoutRoute = true
  requiresHeader = true
  requiresFooter = true

  getRequiredPayloads = (payloads: AllPayloadsType): RequiredPayloadsType =>
    ({disclaimer: payloads.disclaimerPayload})

  getLanguageChangePath = ({location, language}) =>
    this.getRoutePath({city: location.payload.city, language})

  getPageTitle = ({t, cityName}) => cityName ? `${t('pageTitles.disclaimer')} - ${cityName}` : null

  getRoutePath = ({city, language}: DisclaimerRouteParamsType): string =>
    `/${city}/${language}/disclaimer`

  getMetaDescription = () => null

  getFeedbackTargetInformation = ({payloads}) => {
    const disclaimer = payloads.disclaimer.data
    return disclaimer ? {id: disclaimer.id} : null
  }
}

export default DisclaimerRouteConfig
