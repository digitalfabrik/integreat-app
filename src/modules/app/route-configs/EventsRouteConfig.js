// @flow

import type { AllPayloadsType } from './RouteConfig'
import { RouteConfig } from './RouteConfig'
import type { Route } from 'redux-first-router'
import {
  createCitiesEndpoint,
  createEventsEndpoint,
  createLocalNewsEndpoint,
  createLanguagesEndpoint,
  EventModel,
  Payload
} from '@integreat-app/integreat-api-client'
import fetchData from '../fetchData'
import { cmsApiBaseUrl } from '../constants/urls'

type EventsRouteParamsType = {|city: string, language: string|}
type RequiredPayloadsType = {|events: Payload<Array<EventModel>>|}

export const EVENTS_ROUTE = 'EVENTS'

/**
 * EventsRoute, matches /augsburg/de/events and /augsburg/de/events/begegnungscafe
 * @type {{path: string, thunk: function(Dispatch, GetState)}}
 */
const eventsRoute: Route = {
  path: '/:city/:language/events/:eventId?',
  thunk: async (dispatch, getState) => {
    const state = getState()
    const { city, language } = state.location.payload

    await Promise.all([
      fetchData(createCitiesEndpoint(cmsApiBaseUrl), dispatch, state.cities),
      fetchData(createEventsEndpoint(cmsApiBaseUrl), dispatch, state.events, { city, language }),
      fetchData(createLocalNewsEndpoint(cmsApiBaseUrl), dispatch, state.news, { city, language }),
      fetchData(createLanguagesEndpoint(cmsApiBaseUrl), dispatch, state.languages, { city, language })
    ])
  }
}
class EventsRouteConfig implements RouteConfig<EventsRouteParamsType, RequiredPayloadsType> {
  name = EVENTS_ROUTE
  route = eventsRoute
  isLocationLayoutRoute = true
  requiresHeader = true
  requiresFooter = true

  getLanguageChangePath = ({ location, payloads, language }) => {
    const { city, eventId } = location.payload
    const events = payloads.events.data
    if (events && eventId) {
      const event = events.find(_event => _event.path === location.pathname)
      return (event && event.availableLanguages.get(language)) || null
    }
    return this.getRoutePath({ city, language })
  }

  getRequiredPayloads = (payloads: AllPayloadsType): RequiredPayloadsType => ({ events: payloads.eventsPayload })

  getPageTitle = ({ t, payloads, cityName, location }) => {
    if (!cityName) {
      return null
    }
    const pathname = location.pathname
    const events = payloads.events.data
    const event = events && events.find(event => event.path === pathname)
    return `${event ? event.title : t('pageTitles.events')} - ${cityName}`
  }

  getRoutePath = ({ city, language }: EventsRouteParamsType): string => `/${city}/${language}/events`

  getMetaDescription = () => null

  getFeedbackTargetInformation = ({ payloads, location }) => {
    const events = payloads.events.data
    const event = events && events.find(event => event.path === location.pathname)
    return event ? { title: event.title } : null
  }
}

export default EventsRouteConfig
