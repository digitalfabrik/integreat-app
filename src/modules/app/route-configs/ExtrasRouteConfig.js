// @flow

import type { AllPayloadsType } from './RouteConfig'
import { RouteConfig } from './RouteConfig'
import {
  createCitiesEndpoint,
  createEventsEndpoint,
  createExtrasEndpoint,
  createLanguagesEndpoint,
  createLocalNewsEndpoint,
  ExtraModel,
  Payload
} from '@integreat-app/integreat-api-client'
import type { Route } from 'redux-first-router'
import fetchData from '../fetchData'
import { cmsApiBaseUrl } from '../constants/urls'

type ExtrasRouteParamsType = {|city: string, language: string|}
type RequiredPayloadsType = {|extras: Payload<Array<ExtraModel>>|}

export const EXTRAS_ROUTE = 'EXTRAS'

/**
 * ExtrasRoute, matches /augsburg/de/extras and /augsburg/de/extras
 * @type {{path: string, thunk: function(Dispatch, GetState)}}
 */
const extrasRoute: Route = {
  path: '/:city/:language/extras/:extraId?',
  thunk: async (dispatch, getState) => {
    const state = getState()
    const { city, language } = state.location.payload

    await Promise.all([
      fetchData(createCitiesEndpoint(cmsApiBaseUrl), dispatch, state.cities),
      fetchData(createEventsEndpoint(cmsApiBaseUrl), dispatch, state.events, { city, language }),
      fetchData(createLocalNewsEndpoint(cmsApiBaseUrl), dispatch, state.news, { city, language }),
      fetchData(createLanguagesEndpoint(cmsApiBaseUrl), dispatch, state.languages, { city, language }),
      fetchData(createExtrasEndpoint(cmsApiBaseUrl), dispatch, state.extras, { city, language })
    ])
  }
}

class ExtrasRouteConfig implements RouteConfig<ExtrasRouteParamsType, RequiredPayloadsType> {
  name = EXTRAS_ROUTE
  route = extrasRoute
  isLocationLayoutRoute = true
  requiresHeader = true
  requiresFooter = true

  getRoutePath = ({ city, language }: ExtrasRouteParamsType): string => `/${city}/${language}/extras`

  getRequiredPayloads = (payloads: AllPayloadsType): RequiredPayloadsType => ({ extras: payloads.extrasPayload })

  getLanguageChangePath = ({ location, language }) =>
    this.getRoutePath({ city: location.payload.city, language })

  getPageTitle = ({ t, cityName }) => cityName ? `${t('pageTitles.extras')} - ${cityName}` : null

  getMetaDescription = () => null

  getFeedbackTargetInformation = () => null
}

export default ExtrasRouteConfig
