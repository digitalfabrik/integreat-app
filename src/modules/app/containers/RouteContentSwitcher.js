// @flow

import * as React from 'react'
import type { LocationState } from 'redux-first-router'
import type { AllPayloadsType } from '../route-configs/RouteConfig'
import { Payload } from '@integreat-app/integreat-api-client'
import find from 'lodash/find'
import LoadingSpinner from '../../common/components/LoadingSpinner'
import FailureSwitcher from '../../common/components/FailureSwitcher'
import { getRouteConfig } from '../route-configs'
import reduce from 'lodash/reduce'
import { getRouteContent } from '../routeContents'
import { SPRUNGBRETT_EXTRA } from '../route-configs/SprungbrettRouteConfig'
import { WOHNEN_EXTRA } from '../route-configs/WohnenRouteConfig'
import ContentNotFoundError from '../../common/errors/ContentNotFoundError'

type PropsType = {|
  location: LocationState,
  allPayloads: AllPayloadsType
|}

class RouteContentSwitcher extends React.PureComponent<PropsType> {
  renderFailureLoadingComponents = (payloads: {[string]: Payload<any>}): React.Node => {
    const errorPayload = find(payloads, payload => payload.error)
    if (find(payloads, payload => (payload.isFetching || !payload.data) && !payload.error)) {
      return <LoadingSpinner />
    } else if (errorPayload) {
      return <FailureSwitcher error={errorPayload.error} />
    }
    return null
  }

  isWaitingForExtra = (extraType: 'offers' | 'sprungbrettJobs', payloads: {[string]: Payload<any>}): boolean => {
    const alias = extraType === 'offers' ? WOHNEN_EXTRA : SPRUNGBRETT_EXTRA
    return payloads[extraType] && !find(payloads.extras.data, extra => extra.alias === alias)
  }

  renderExtraFailure = (payloads: {[string]: Payload<any>}, location: LocationState): React.Node => {
    if (payloads.extras && !payloads.extras.isFetching && payloads.extras.data && !payloads.extras.error) {
      if (this.isWaitingForExtra('offers', payloads) || this.isWaitingForExtra('sprungbrettJobs', payloads)) {
        return <FailureSwitcher error={new ContentNotFoundError({
          type: 'extra',
          id: location.pathname,
          city: location.payload.city,
          language: location.payload.language
        })} />
      }
    }
    return null
  }

  render () {
    const { location, allPayloads } = this.props
    const currentRoute = location.type
    const RouteContent = getRouteContent(currentRoute)
    const payloads = getRouteConfig(currentRoute).getRequiredPayloads(allPayloads)
    return this.renderExtraFailure(payloads, location) ||
      this.renderFailureLoadingComponents(payloads) ||
      <RouteContent {...reduce(payloads, (result, value, key: string) => ({ [key]: value.data, ...result }), {})} />
  }
}

export default RouteContentSwitcher
