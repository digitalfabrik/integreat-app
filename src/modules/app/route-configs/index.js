// @flow

import type { Route } from 'redux-first-router'
import { RouteConfig } from './RouteConfig'
import CategoriesRouteConfig, { CATEGORIES_ROUTE } from './CategoriesRouteConfig'
import I18nRedirectRouteConfig, { I18N_REDIRECT_ROUTE } from './I18nRedirectRouteConfig'
import LandingRouteConfig, { LANDING_ROUTE } from './LandingRouteConfig'
import MainDisclaimerRouteConfig, { MAIN_DISCLAIMER_ROUTE } from './MainDisclaimerRouteConfig'
import EventsRouteConfig, { EVENTS_ROUTE } from './EventsRouteConfig'
import NewsRouteConfig, { NEWS_ROUTE } from './NewsRouteConfig'
import LocalNewsDetailsRouteConfig, { LOCAL_NEWS_DETAILS_ROUTE } from './LocalNewsDetailsRouteConfig'
import TuNewsDetailsRouteConfig, { TUNEWS_DETAILS_ROUTE } from './TuNewsDetailsRouteConfig'
import TuNewsListRouteConfig, { TUNEWS_LIST_ROUTE } from './TuNewsListRouteConfig'
import WohnenRouteConfig, { WOHNEN_ROUTE } from './WohnenRouteConfig'
import ExtrasRouteConfig, { EXTRAS_ROUTE } from './ExtrasRouteConfig'
import SearchRouteConfig, { SEARCH_ROUTE } from './SearchRouteConfig'
import PoisRouteConfig, { POIS_ROUTE } from './PoisRouteConfig'
import DisclaimerRouteConfig, { DISCLAIMER_ROUTE } from './DisclaimerRouteConfig'
import SprungbrettRouteConfig, { SPRUNGBRETT_ROUTE } from './SprungbrettRouteConfig'
import NotFoundRouteConfig from './NotFoundRouteConfig'
import { NOT_FOUND } from 'redux-first-router'

const routeConfigs: {[string]: RouteConfig<any, any>} = {
  [MAIN_DISCLAIMER_ROUTE]: new MainDisclaimerRouteConfig(),
  [I18N_REDIRECT_ROUTE]: new I18nRedirectRouteConfig(),
  [LANDING_ROUTE]: new LandingRouteConfig(),
  [EVENTS_ROUTE]: new EventsRouteConfig(),
  [NEWS_ROUTE]: new NewsRouteConfig(),
  [LOCAL_NEWS_DETAILS_ROUTE]: new LocalNewsDetailsRouteConfig(),
  [TUNEWS_DETAILS_ROUTE]: new TuNewsDetailsRouteConfig(),
  [TUNEWS_LIST_ROUTE]: new TuNewsListRouteConfig(),
  [SPRUNGBRETT_ROUTE]: new SprungbrettRouteConfig(),
  [WOHNEN_ROUTE]: new WohnenRouteConfig(),
  [EXTRAS_ROUTE]: new ExtrasRouteConfig(),
  [DISCLAIMER_ROUTE]: new DisclaimerRouteConfig(),
  [SEARCH_ROUTE]: new SearchRouteConfig(),
  [POIS_ROUTE]: new PoisRouteConfig(),
  [CATEGORIES_ROUTE]: new CategoriesRouteConfig(),
  [NOT_FOUND]: new NotFoundRouteConfig(),
}

export const getRouteConfig = (routeName: string): RouteConfig<*, *> => {
  const routeConfig = routeConfigs[routeName]
  if (!routeConfig) {
    throw new Error(
      `There is no route config with the name ${routeName}. Did you forget to add it in the routes index?`)
  }
  return routeConfig
}

export const routesMap: {[string]: Route} = {
  [MAIN_DISCLAIMER_ROUTE]: new MainDisclaimerRouteConfig().route,
  [I18N_REDIRECT_ROUTE]: new I18nRedirectRouteConfig().route,
  [LANDING_ROUTE]: new LandingRouteConfig().route,
  [EVENTS_ROUTE]: new EventsRouteConfig().route,
  [NEWS_ROUTE]: new NewsRouteConfig().route,
  [LOCAL_NEWS_DETAILS_ROUTE]: new LocalNewsDetailsRouteConfig().route,
  [TUNEWS_DETAILS_ROUTE]: new TuNewsDetailsRouteConfig().route,
  [TUNEWS_LIST_ROUTE]: new TuNewsListRouteConfig().route,
  [SPRUNGBRETT_ROUTE]: new SprungbrettRouteConfig().route,
  [WOHNEN_ROUTE]: new WohnenRouteConfig().route,
  [EXTRAS_ROUTE]: new ExtrasRouteConfig().route,
  [DISCLAIMER_ROUTE]: new DisclaimerRouteConfig().route,
  [SEARCH_ROUTE]: new SearchRouteConfig().route,
  [POIS_ROUTE]: new PoisRouteConfig().route,
  [CATEGORIES_ROUTE]: new CategoriesRouteConfig().route,
}
