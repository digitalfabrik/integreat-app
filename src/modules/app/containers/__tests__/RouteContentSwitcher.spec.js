// @flow

import RouteContentSwitcher from '../RouteContentSwitcher'
import { CategoriesMapModel, CategoryModel, CityModel, Payload } from '@integreat-app/integreat-api-client'
import moment from 'moment'
import { mount, shallow } from 'enzyme'
import React from 'react'
import createLocation from '../../../../createLocation'
import { CATEGORIES_ROUTE } from '../../route-configs/CategoriesRouteConfig'
import LoadingSpinner from '../../../common/components/LoadingSpinner'
import theme from '../../../theme/constants/theme'
import { ThemeProvider } from 'styled-components'

describe('RouteContentSwitcher', () => {
  const categories = new CategoriesMapModel([
    new CategoryModel({
      root: true,
      path: 'path01',
      title: 'Title10',
      content: 'contnentl',
      thumbnail: 'thumb/nail',
      parentPath: 'parent/url',
      order: 4,
      availableLanguages: new Map(),
      lastUpdate: moment('2017-11-18T09:30:00.000Z'),
      hash: '2fe6283485a93932'
    })
  ])

  const cities = [
    new CityModel({
      name: 'Mambo No. 5',
      code: 'city1',
      live: true,
      eventsEnabled: true,
      extrasEnabled: false,
      pushNotificationsEnabled: false,
      tunewsEnabled: false,
      sortingName: 'Mambo'
    })
  ]

  const categoriesPayload = new Payload(false, 'https://random.api.json', categories, null)
  const citiesPayload = new Payload(false, 'https://random.api.json', cities, null)
  const fetchingPayload = new Payload(true)
  const errorPayload = new Payload(false, 'https://random.api.json', null, new Error('error'))
  const allPayloads = {
    categoriesPayload,
    citiesPayload,
    poisPayload: fetchingPayload,
    eventsPayload: fetchingPayload,
    localNewsPayload: fetchingPayload,
    localNewsElementPayload: fetchingPayload,
    tunewsPayload: fetchingPayload,
    tunewsElementPayload: fetchingPayload,
    extrasPayload: fetchingPayload,
    sprungbrettJobsPayload: fetchingPayload,
    wohnenPayload: fetchingPayload,
    disclaimerPayload: fetchingPayload
  }

  const fetchingPayloads = {
    categoriesPayload: fetchingPayload,
    citiesPayload: fetchingPayload,
    poisPayload: fetchingPayload,
    eventsPayload: fetchingPayload,
    localNewsPayload: fetchingPayload,
    localNewsElementPayload: fetchingPayload,
    tunewsPayload: fetchingPayload,
    tunewsElementPayload: fetchingPayload,
    extrasPayload: fetchingPayload,
    sprungbrettJobsPayload: fetchingPayload,
    wohnenPayload: fetchingPayload,
    disclaimerPayload: fetchingPayload
  }

  it('should render a FailureSwitcher if a payload contains an error', () => {
    const location = createLocation({ type: CATEGORIES_ROUTE, payload: { city: 'augsburg', language: 'de' } })
    const renderFailureLoadingComponents = shallow(
      <RouteContentSwitcher allPayloads={allPayloads} location={location} />
    ).instance().renderFailureLoadingComponents

    expect(renderFailureLoadingComponents({ payload: errorPayload })).toMatchSnapshot()
  })

  it('should render a Spinner if data has not been fetched yet', () => {
    const location = createLocation({ type: CATEGORIES_ROUTE, payload: { city: 'augsburg', language: 'de' } })

    expect(mount(
        <ThemeProvider theme={theme}>
          <RouteContentSwitcher allPayloads={fetchingPayloads} location={location} />
        </ThemeProvider>
    ).find(LoadingSpinner)).not.toBeUndefined()
  })

  it('should render and match snapshot', () => {
    const location = createLocation({ type: CATEGORIES_ROUTE, payload: { city: 'augsburg', language: 'de' } })

    expect(shallow(<RouteContentSwitcher location={location} allPayloads={allPayloads} />)).toMatchSnapshot()
  })
})
