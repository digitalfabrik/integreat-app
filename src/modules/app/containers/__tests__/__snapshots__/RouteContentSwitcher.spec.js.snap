// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteContentSwitcher should render a FailureSwitcher if a payload contains an error 1`] = `
<FailureSwitcher
  error={[Error: error]}
/>
`;

exports[`RouteContentSwitcher should render and match snapshot 1`] = `
<Connect(CategoriesPage)
  categories={
    CategoriesMapModel {
      "_categories": Map {
        "path01" => CategoryModel {
          "_availableLanguages": Map {},
          "_content": "contnentl",
          "_hash": "2fe6283485a93932",
          "_lastUpdate": "2017-11-18T09:30:00.000Z",
          "_order": 4,
          "_parentPath": "parent/url",
          "_path": "path01",
          "_root": true,
          "_thumbnail": "thumb/nail",
          "_title": "Title10",
        },
      },
    }
  }
  cities={
    Array [
      CityModel {
        "_aliases": undefined,
        "_code": "city1",
        "_eventsEnabled": true,
        "_extrasEnabled": false,
        "_latitude": undefined,
        "_live": true,
        "_longitude": undefined,
        "_name": "Mambo No. 5",
        "_prefix": undefined,
        "_pushNotificationsEnabled": false,
        "_sortingName": "Mambo",
        "_tunewsEnabled": false,
      },
    ]
  }
/>
`;
